charts/consul/templates/ingress-gateways-deployment.yaml

```
@@ -107,374 +146,165 @@ spec:
+          {{- end }}
+          items:
+          - key: {{ default "tls.crt" $root.Values.global.tls.caCert.secretKey }}
+            path: tls.crt
+      {{- end }}
+      {{- end }}
+      initContainers:
+      # ingress-gateway-init registers the ingress gateway service with Consul.
+      - name: ingress-gateway-init
+        image: {{ $root.Values.global.imageK8S }}
+        env:
+        - name: NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        {{- include "consul.consulK8sConsulServerEnvVars" $root | nindent 8 }}
+        {{- if $root.Values.global.acls.manageSystemACLs }}
+        - name: CONSUL_LOGIN_AUTH_METHOD
+          value: {{ template "consul.fullname" $root }}-k8s-component-auth-method
+        - name: CONSUL_LOGIN_DATACENTER
+          value: {{ $root.Values.global.datacenter }}
+        - name: CONSUL_LOGIN_META
+          value: "component=ingress-gateway,pod=$(NAMESPACE)/$(POD_NAME)"
+       {{- end }}
+        command:
+        - "/bin/sh"
+        - "-ec"
+        - |
+          consul-k8s-control-plane connect-init -pod-name=${POD_NAME} -pod-namespace=${NAMESPACE} \
+          -gateway-kind="ingress-gateway" \
+          -consul-node-name="k8s-service-mesh" \
+          -proxy-id-file=/consul/service/proxy-id \
+          -service-name={{ template "consul.fullname" $root }}-{{ .name }} \
+          -log-level={{ default $root.Values.global.logLevel }} \
+          -log-json={{ $root.Values.global.logJSON }}
+        volumeMounts:
         - name: consul-service
-          emptyDir:
-            medium: "Memory"
+          mountPath: /consul/service
         {{- if $root.Values.global.tls.enabled }}
-        {{- if not (and $root.Values.externalServers.enabled $root.Values.externalServers.useSystemRoots) }}
         - name: consul-ca-cert
-          secret:
-            {{- if $root.Values.global.tls.caCert.secretName }}
-            secretName: {{ $root.Values.global.tls.caCert.secretName }}
-            {{- else }}
-            secretName: {{ template "consul.fullname" $root }}-ca-cert
-            {{- end }}
-            items:
-            - key: {{ default "tls.crt" $root.Values.global.tls.caCert.secretKey }}
-              path: tls.crt
-        {{- end }}
-        {{- if $root.Values.global.tls.enableAutoEncrypt }}
-        - name: consul-auto-encrypt-ca-cert
-          emptyDir:
-            medium: "Memory"
+          mountPath: /consul/tls/ca
+          readOnly: true
         {{- end }}
+        resources:
+          requests:
+            memory: "50Mi"
+            cpu: "50m"
+          limits:
+            memory: "50Mi"
+            cpu: "50m"
+      containers:
+      - name: ingress-gateway
+        image: {{ $root.Values.global.imageConsulDataplane | quote }}
+        {{- if (default $defaults.resources .resources) }}
```

```suggestion
        {{- if (default $defaults.resources $root.resources) }}
```
Does the non-default value need to be pinned to `$root` here and below?

---

Its only those that start with a `.`